src/Build/BackEnd/Node/OutOfProcNode.cs

```diff
@@ -472,11 +472,18 @@ private NodeEngineShutdownReason HandleShutdown(out Exception exception)
             // so reset it away from a user-requested folder that may get deleted.
             NativeMethodsShared.SetCurrentDirectory(BuildEnvironmentHelper.Instance.CurrentMSBuildToolsDirectory);
 
-            // Restore the original environment.
+            // Restore the original environment, best effort.
             // If the node was never configured, this will be null.
             if (_savedEnvironment != null)
             {
-                CommunicationsUtilities.SetEnvironment(_savedEnvironment);
+                try
+                {
+                    CommunicationsUtilities.SetEnvironment(_savedEnvironment);
```

Comment:
I could add a `NoThrow` wrapper but the callers are interested in logging the exception so that would have to be something like `Exception SetEnvironmentNoThrow(...)` which would be an awkward method signature. Maybe I am misunderstanding the suggestion. Please elaborate.

Reply:
I forget to mention it is NIT.
I though in line of `SetEnvironment(IEnumerable<...> envVars, bool forgiving = false);` current usage would not change only places where we require to be forgiving you would add `forgiving: true` which would swallow and trace-log all those `Set` exceptions.

Reply:
The two callers with `try`/`catch` unfortunately don't use the same trace-logging mechanism. That itself may not be done for a good reason but I'd prefer to carefully step over this rabbit hole and merge as is.